gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -472,26 +472,7 @@ private void addOrAlterPartition(IMetaStoreClient client, Table table, HiveParti
             table.getTableName(), nativePartition.getSd().getLocation()));
       } catch (TException e) {
         try {
-          HivePartition existingPartition;
-          try (Timer.Context context = this.metricContext.timer(GET_HIVE_PARTITION).time()) {
-            existingPartition = HiveMetaStoreUtils.getHivePartition(
-                client.getPartition(table.getDbName(), table.getTableName(), nativePartition.getValues()));
-          }
-
-          if (needToUpdatePartition(existingPartition, partition)) {
-            log.info(String.format("Partition update required. ExistingPartition %s, newPartition %s",
-                stringifyPartition(existingPartition), stringifyPartition(partition)));
-            Partition newPartition = getPartitionWithCreateTime(nativePartition, existingPartition);
-            log.info(String.format("Altering partition %s", newPartition));
-            try (Timer.Context context = this.metricContext.timer(ALTER_PARTITION).time()) {
-              client.alter_partition(table.getDbName(), table.getTableName(), newPartition);
-            }
-            log.info(String.format("Updated partition %s in table %s with location %s", stringifyPartition(newPartition),
-                table.getTableName(), nativePartition.getSd().getLocation()));
-          } else {
-            log.info(String.format("Partition %s in table %s with location %s already exists and no need to update",
-                stringifyPartition(nativePartition), table.getTableName(), nativePartition.getSd().getLocation()));
-          }
+          OnPartitionExist(client, table, partition, nativePartition, e);
```

Comment:
we typically adopt the convention of having variable names and methods start with lower case. so onPartitionExists(...)?